### PR TITLE
GP tutorial NumPy error

### PR DESCRIPTION
**Pull request by ndmlny-qs**
The `Gaussian_Process_Gpytorch` tutorial has been updated for conversion
to MDX. See #1392 for more info about the updates. Only minor edits were
made including:

- link updates
- markdown display of dataframes
- using ArviZ for summary statistics
- spelling and grammar checks

In cells 12 and 13, we attempt to show the summary statistics from ArviZ
for the model. The following error occurs when converting the Bean
Machine `MonteCarloSamples` object to either an xarray or ArviZ
`InferenceData` object.

```python
RuntimeError: Can't call numpy() on Tensor that requires grad. Use
tensor.detach().numpy() instead.
```

This error is coming from the `to_inference_data` method, and can be
seen in the stack trace in the notebook.

### Types of changes
- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](https://github.com/facebookresearch/beanmachine/blob/main/CONTRIBUTING.md)** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by ndmlny-qs** [comment]
Note that the PR was created so reviewers can investigate the stack traces in the notebook showing the error.

**Comment by jpchen** [comment]
After discussing with @horizon-blue, we should make a PR in arviz to detach the Pytorch tensors before converting, but in the meantime, we can detach them in MonteCarloSamples.

**Comment by ndmlny-qs** [comment]
@jpchen & @horizon-blue I've updated this PR to include the detaching of pytorch tensors in the `to_inference_data` method.

**Comment by jpchen** [comment]
Thanks for making the fix! This LGTM assuming the GP tutorial passes (not being tested in our CI currently unfortunately). Not related to this diff, but can you also remove the ['tutorials landing page'](https://beanmachine.org/docs/tutorials/) now since they are now on the website natively?

**Comment by ndmlny-qs** [comment]
@jpchen would you like the landing page to be the coin flipping tutorial, or do you want me to modify the landing page to link to the rendered tutorials?

**Comment by jpchen** [comment]
I think coin flipping is fine, fewer places to maintain if/when we add more tutorials

**Comment by ndmlny-qs** [comment]
done, we can revert back if we want, and I can make an automatically generated landing page based off of the ipynb2mdx converter.

**Comment by jpchen** [comment]
LGTM, thanks for making the changes!

**Comment by horizon-blue** [comment]
Ah, our internal static site builder doesn't know the custom script to render the tutorials, so it errors out :).

> I think coin flipping is fine, fewer places to maintain if/when we add more tutorials

@jpchen  If we want to keep the maintenance effort low, I wonder if we should create a generic landing page (e.g. one that simply says "check the sidebar for details", or something like [Pyro's tutorial home page](https://pyro.ai/examples/))?

**Comment by jpchen** [comment]
> if we should create a generic landing page (e.g. one that simply says "check the sidebar for details", or something like [Pyro's tutorial home page](https://pyro.ai/examples/))?

Pyro's tutorial home page is really more analogous to [our main page](https://beanmachine.org/docs/overview/why_bean_machine/) since the whole site contains everything about system design and intro to Bayesian modeling/VI, not just notebooks. Since we have all the other exposition nicely broken up into headers in the sidebar, I feel like the specific tutorials page is a bit redundant. If you think there's valuable information we can share on that page, I'm happy to leave it/repurpose it, but as is, is not very useful IMO.  Will follow up with you on the internal build failure, not quite clear to me yet how that's related.

File: src/beanmachine/ppl/inference/monte_carlo_samples.py [hunk 1]
```diff
@@ -273,8 +273,15 @@ def to_inference_data(self, include_adapt_steps: bool = False) -> az.InferenceDa
 
         if "posterior" in self.namespaces:
             posterior = self.namespaces["posterior"].samples
+            posterior = {
+                key: value.detach().numpy() for key, value in posterior.items()
+            }
```
Comment (horizon-blue) [inline src/beanmachine/ppl/inference/monte_carlo_samples.py:278]: Can we refactor this into a utility function so we don't need to repeat the same set of code for each field?

**Review by horizon-blue — COMMENTED** [review]
Sorry for being late to the party. Our internal static doc building issue should've been resolved in https://github.com/facebookresearch/beanmachine/pull/1410. The internal tests should pass once we reimport the PR. :) I just have a small comment regarding the section where we detach the tensors. Other than that this PR looks great to me!

File: src/beanmachine/ppl/inference/monte_carlo_samples.py (hunk 1 above)
Comment (ndmlny-qs) [reply src/beanmachine/ppl/inference/monte_carlo_samples.py]: Created a new utility method, and created a new test for it.

**Comment by feynmanliang** [comment]
Warnings are unrelated, I'll go ahead and kick of the landing process for this :)

**Pull request closed without merging**